src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs

```diff
@@ -21,7 +21,8 @@ internal enum ExpansionKind
         RawView,
         ResultsView,
         StaticMembers,
-        TypeVariable
+        TypeVariable,
+        ExplicitEvaluationResult,
```

Comment:
Do we need both a Kind and an "ExternalEvaluationResult" property?  It seems it might be better to just have the property and null check it.  Currently, most usages seem to check the Kind and assume the property is set.  However, ```GetResult``` null checks without checking the Kind.

Naively, it also seems like it might be better to just leave ```CreateDataItem``` as is and override ```GetResult``` instead (but I might be missing something).